File: leetcode/70.ClimbingStairs.py
class Solution:
    @cache
    def dfs(self, start: int, target: int) -> int:
        if start == target: return 1
        if start > target:  return 0
        return self.dfs(start+1, target) + self.dfs(start+2, target)

    def climbStairs(self, n: int) -> int:
        return self.dfs(0, n)

File: leetcode/70.ClimbingStairs.cpp
#include <bits/stdc++.h>
#define ll long long
using namespace std;
const int mx = 1e6 + 3;

vector<ll> memo(mx, -1);
ll sum = 0;

ll calcWays(ll start, ll target){
  if(start > target){ return 0; }
  if(start == target){ return 1; }
  if(memo[start] != -1){ return memo[start]; }

  memo[start] = calcWays(start+1, target) + calcWays(start+2, target);
  return memo[start];
}

int main(){
  ll n;
  cin >> n;
  sum = calcWays(0, n);

  cout << sum << endl;
}